File: Baekjoon/1259.py
import sys

while True:
    tmp = sys.stdin.readline().rstrip()
    if tmp == '0':
        break
    left = 0
    right = len(tmp) - 1
    is_palindrome = True
    while left <= right:
        if tmp[left] != tmp[right]:
            print("no")
            is_palindrome = False
            break
        left += 1
        right -= 1
    if is_palindrome:
        print("yes")

File: Baekjoon/1259.cpp
#include <iostream>
using namespace std;

int main() {
  while (true) {
    string num;
    cin >> num;
    if (num == "0") break;

    int s = 0, e = num.length() - 1;
    bool is_palindrome = true;
    while(s < e) {
      if(num[s] != num[e]){
        is_palindrome = false;
        break;
      }
      s++;
      e--;
    }
    if(is_palindrome) cout << "yes\n";
    else cout << "no\n";
  }
  return 0;
}